typings/preact/helpers.d.ts

```diff
@@ -1,5 +1,7 @@
-type Diff<T extends string, U extends string> = ({[P in T]: P} &
-  {[P in U]: never} & {[x: string]: never})[T]
+type Diff<
+  T extends string | number | symbol,
+  U extends string | number | symbol
+> = ({[P in T]: P} & {[P in U]: never} & {[x: string]: never})[T]
```

Comment:
However, after fix this, Glamorous typing will not work with TS <= 2.8

Reply:
Ah, have you tested that? I'm guessing the `in T` and `in U` don't work? this is why flow's approach is better (versioning to lib and flow instead of just lib). Any way, it seems like we should prefer supporting the most recent version of TS, no?

Reply:
@aaronjensen As you said, supporting recent TS version is better. However, if there's a way to support both old and new versions, that's the best, isn't it?
I mean, `keyof any` will return `string | number | symbol` for TS@2.9, and `string` for TS@<=2.8, so how about that?

Reply:
And if we want to support TS@2.9 only, we have `Extract` from TS@2.8, so we don't need these helpers.